Pear/Profile/ProfileViewController.swift

```diff
@@ -62,37 +73,95 @@ class ProfileViewController: UIViewController {
+        view.addGestureRecognizer(tapGesture)
+        
         setupConstraints()
     }
-
-    private func getOtherUser(otherUserId: Int) {
-        NetworkManager.getUser(id: otherUserId) { [weak self] result in
+    
+    private func getProfileUser(profileUserId: Int) {
+        NetworkManager.getUser(id: profileUserId) { [weak self] result in
             guard let self = self else { return }
             switch result {
             case .success(let user):
                 DispatchQueue.main.async {
-                    self.otherUser = user
-                    self.profileSections = [.summary, .basics]
-                    
-                    if !user.interests.isEmpty {
-                        self.profileSections.append(.interests)
-                    }
-                    
-                    if !user.groups.isEmpty {
-                        self.profileSections.append(.groups)
-                    }
-                    
-                    if !user.prompts.isEmpty {
-                        self.profileSections.append(.prompts)
+                    self.profileUser = user
+                    if let blocked = self.profileUser?.isBlocked as? Bool, blocked {
+                        self.setupBlockedProfile()
+                    } else {
+                        self.setupProfile()
                     }
-                    
                     self.profileTableView.reloadData()
                 }
             case .failure(let error):
                 print(error.localizedDescription)
             }
         }
     }
+            
+    private func setupBlockedProfile() {
+        profileSections = [.summary]
+        menuButton.isHidden = true
+        unblockButton.isHidden = false
+        unblockLabel.isHidden = false
+        profileTableView.isScrollEnabled = false
+        
+        unblockButton.setTitle("Unblock", for: .normal)
+        unblockButton.setTitleColor(.white, for: .normal)
+        unblockButton.titleLabel?.font = ._20CircularStdBold
+        unblockButton.layer.cornerRadius = Constants.Onboarding.mainButtonSize.height / 2
+        unblockButton.backgroundColor = .backgroundOrange
+        unblockButton.isEnabled = true
+        unblockButton.addTarget(self, action: #selector(unblockedButtonPressed), for: .touchUpInside)
+        view.addSubview(unblockButton)
+        
+        unblockLabel.text = "This user is currently blocked."
+        unblockLabel.font = ._18CircularStdBook
+        unblockLabel.textAlignment = .center
+        unblockLabel.numberOfLines = 0
+        unblockLabel.textColor = .darkGreen
+        view.addSubview(unblockLabel)
+        
+        let unblockLabelSize = CGSize(width: 280, height: 25)
+        let unblockButtonSize = CGSize(width: 196, height: 53)
+        
+        unblockLabel.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.centerY.equalToSuperview().offset(50)
+            make.size.equalTo(unblockLabelSize)
```

Comment:
why is this setting the size for labels? I feel like it could cut off the label for certain screen sizes? labels with only an x and y and will auto-size themselves to fit the text